### PR TITLE
Upgrade to django-registration-redux 1.2

### PR DESCRIPTION
**Pull request by jamesmcclain**
django-registration-redux has been upgraded from 1.1 to 1.2.  The
workarounds for short-comings in the former have been removed.

Connects #156

I did the following:
   * Applied the changes in the patch
   * Destroyed the app and services VMs
   * Did a `vagrant up` which recreated the VMs
   * Ran migrations and reload_dev_data.  The migrations succeeded and there were no errors or warnings.
   * Ran tests: `./scripts/manage test` and they all succeeded
   * Created new users, logged in, and so on

### CODE REVIEW

**Comment by caseypt** [comment]
Sorry, my mention of the work-around didn't have the whole work-around highlighted. The [next few lines](https://github.com/WikiWatershed/model-my-watershed/blob/develop/src/mmw/mmw/settings/base.py#L338-L340) below what I highlighted are also part of it.

We should be able to remove that setting and the attached migration. The initial migration is [now provided](https://github.com/macropin/django-registration/releases) as part of the package: 

> BACKWARDS-INCOMPATIBLE CHANGE: An initial migration for Django > 1.7 has been
provided. South users will need to configure a null migration with
(SOUTH_MIGRATION_MODULES).

**Comment by jamesmcclain** [comment]
No problem, thanks for the clarification.  I will try again.

**Comment by caseypt** [comment]
The associated migration can be found here: https://github.com/WikiWatershed/model-my-watershed/tree/develop/src/mmw/mmw/migrations

**Comment by rajadain** [comment]
When I try to sign up as a new user, I get this error:

```
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/base.py", line 132, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python2.7/dist-packages/django/views/decorators/csrf.py", line 58, in wrapped_view
    return view_func(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/django/views/generic/base.py", line 71, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/rest_framework/views.py", line 452, in dispatch
    response = self.handle_exception(exc)
  File "/usr/local/lib/python2.7/dist-packages/rest_framework/views.py", line 449, in dispatch
    response = handler(request, *args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/rest_framework/decorators.py", line 50, in handler
    return func(*args, **kwargs)
  File "/opt/app/apps/user/views.py", line 182, in sign_up
    user = view.register(request, **form.cleaned_data)
TypeError: register() got an unexpected keyword argument 'username'
[08/Jul/2015 11:13:50]"POST /user/sign_up HTTP/1.0" 500 12195
```

**Comment by jamesmcclain** [comment]
Corrected, thank you.

**Comment by rajadain** [comment]
+1 works great! Tested:

 * Signing Up as a new user
 * Resetting password
 * Retrieving forgotten username with email
 * Resending activation email

I did notice a typo from earlier work here: https://github.com/WikiWatershed/model-my-watershed/blob/develop/src/mmw/js/src/user/templates/signUpModal.html#L5

screen shot 2015-07-09 at 10 07 49 am

Could you please change "active" to "activate"?

File: src/mmw/apps/user/views.py [hunk 1]
```diff
@@ -179,7 +179,7 @@ def sign_up(request):
     form = RegistrationFormUniqueEmail(request.POST)
 
     if form.is_valid():
-        user = view.register(request, **form.cleaned_data)
+        user = view.register(request, form)
```
Comment (rajadain) [inline src/mmw/apps/user/views.py]: Nice fix, this wasn't really documented in their "BREAKING CHANGES" section.

**Comment by jamesmcclain** [comment]
Certainly, will-do.

**Comment by jamesmcclain** [comment]
According to the diff, there are three functions which have undergone signature changes, and `register` is the only one that we appear to be using.

**Comment by caseypt** [comment]
Reviewing now.

**Comment by caseypt** [comment]
Not a big deal, but the entire `src/mmw/mmw/migrations` folder can be deleted now.

**Comment by caseypt** [comment]
+1. Tested. Nice job.

**Comment by jamesmcclain** [comment]
Okay, I will delete that directory then merge.  Thank you for testing.

**Pull request merged**